job_definitions/validate_cloudtrail_logs.yml

```diff
@@ -0,0 +1,92 @@
+---
+- job:
+    name: validate-cloudtrail-logs
+    project-type: freestyle
+    display-name: Validate logs stored by Amazon CloudWatch in S3.
+    parameters:
+      - choice:
+          name: ACCOUNT
+          choices:
+            - main
+            - development
+            - production
+            - backups
+      - string:
+          name: START_FROM
+          default: "3 days ago"
+          description: Logs delivered on or after the specified time will be validated. Can be in any format accepted by GNU date.
+      - string:
+          name: UP_TO
+          default: "now"
+          description: Logs delivered on or before the specified time will be validated. Can be in any format accepted by GNU date.
+      - bool:
+          name: VERBOSE
+          default: false
+          description: Print verbose log validation information.
+    builders:
+      - shell: |
+          docker run --rm \
+            -v /var/lib/jenkins/digitalmarketplace-credentials:/digitalmarketplace-credentials \
+            -e DM_CREDENTIALS_REPO=/digitalmarketplace-credentials \
+            -e VERBOSE=$VERBOSE \
+            digitalmarketplace/scripts \
+            scripts/validate-cloudtrail-logs.sh $ACCOUNT $START_FROM $UP_TO
+
```

Comment:
Think it would be cool to do this as a loop for the different accounts. On the off chance it fails then it won't fail for all and will indicate the correct account.

Reply:
@benvand There are two kind of "failures", one is when validation is not performed for any reason (permissions, connectivity, etc.), while the other is when validation returns at least one invalid log or digest file. In both cases these are caught and stored below, telling us which account failed for us to check manually. To my understanding if one stage fails the rest will be checked, but I could be wrong.